File: app/src/main/java/com/neome/api/meta/base/dto/DefnSection.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Types.EnumDefnPlacement;
import com.neome.api.meta.base.Types.EnumDefnThemeDirection;
import com.neome.api.meta.base.Types.EnumDefnThemeSectionVariant;
import com.neome.api.meta.base.Types.MetaIdField;
import com.neome.api.meta.base.Types.MetaIdSection;

import org.jetbrains.annotations.Nullable;

import java.util.Map;

@SuppressWarnings("unused")
public class DefnSection extends DefnComp
{
  @Nullable
  public DefnStudioMapOfActionPermission actionPermissionMap;

  @Nullable
  public MetaIdField[] fieldIdSet;

  @Nullable
  public Map<MetaIdField, Long> fieldSpanMap;

  @Nullable
  public Boolean flexGrow;

  @Nullable
  public EnumDefnPlacement justifyContent;

  public MetaIdSection metaId;

  @Nullable
  public Boolean overflowHidden;

  @Nullable
  public String propertyEditorLabel;

  @Nullable
  public FieldDtoSectionLayout reportLayout;

  @Nullable
  public EnumDefnThemeDirection sectionDirection;

  @Nullable
  public EnumDefnThemeSectionVariant sectionVariant;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnSection.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumDefnPlacement
import com.neome.api.meta.base.Types.EnumDefnThemeDirection
import com.neome.api.meta.base.Types.EnumDefnThemeSectionVariant
import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdSection

interface DefnSection : DefnComp {
    val actionPermissionMap: DefnStudioMapOfActionPermission?
    val fieldIdSet: List<MetaIdField>?
    val fieldSpanMap: Map<MetaIdField, Long>?
    val flexGrow: Boolean?
    val justifyContent: EnumDefnPlacement?
    val metaId: MetaIdSection
    val overflowHidden: Boolean?
    val propertyEditorLabel: String?
    val reportLayout: FieldDtoSectionLayout?
    val sectionDirection: EnumDefnThemeDirection?
    val sectionVariant: EnumDefnThemeSectionVariant?
}
